src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -68,7 +69,69 @@ public function getItem(string $resourceClass, $id, string $operationName = null
             throw new ResourceClassNotSupportedException();
         }
 
-        $identifierValues = explode('-', $id);
+        $identifiers = $this->normalizeIdentifiers($id, $manager, $resourceClass);
+
+        if (!$fetchData && $manager instanceof EntityManagerInterface) {
+            return $manager->getReference($resourceClass, $identifiers);
+        }
+
+        $repository = $manager->getRepository($resourceClass);
+        $queryBuilder = $repository->createQueryBuilder('o');
+
+        $this->addWhereForIdentifiers($identifiers, $queryBuilder);
+
+        foreach ($this->itemExtensions as $extension) {
+            $extension->applyToItem($queryBuilder, $resourceClass, $identifiers, $operationName);
+        }
+
+        return $queryBuilder->getQuery()->getOneOrNullResult();
+    }
+
+    /**
+     * Add where into the query when multiple identifiers are passed (composite).
+     *
+     * @param array        $identifiers
+     * @param QueryBuilder $queryBuilder
+     *
+     * Populate the query with where when needed.
+     */
+    private function addWhereForIdentifiers(array $identifiers, QueryBuilder $queryBuilder)
+    {
+        if (empty($identifiers)) {
+            return;
+        }
+
+        foreach ($identifiers as $identifier => $value) {
+            $placeholder = ':id_'.$identifier;
+            $expression = $queryBuilder->expr()->eq(
+                'o.'.$identifier,
+                $placeholder
+            );
+
+            $queryBuilder->andWhere($expression);
+
+            $queryBuilder->setParameter($placeholder, $value);
+        }
+    }
+
+    /**
+     * Transform and check the identifier, composite or not.
+     *
+     * @param $id
+     * @param $manager
+     * @param $resourceClass
+     *
+     * @return array
+     */
+    private function normalizeIdentifiers($id, $manager, $resourceClass) : array
+    {
+        $doctrineMetadataIdentifier = $manager->getClassMetadata($resourceClass)->getIdentifier();
+        $identifierValues = [$id];
+
+        if (count($doctrineMetadataIdentifier) >= 2) {
+            $identifierValues = explode('-', $id);
```

Comment:
I still think this is too simplistic and we should take https://github.com/api-platform/core/issues/530#issuecomment-217778102 and https://github.com/api-platform/core/issues/530#issuecomment-217780036 into account.